### PR TITLE
Fix roundrobin overflow bug.

### PR DESCRIPTION
**Pull request by reterVision**

### CODE REVIEW

**Comment by reterVision** [comment]
@benschw any feedbacks on this?

**Comment by benschw** [comment]
sorry for not getting back to you sooner.

I'm a little confused though, what is problem is this solving? does taking the modulus account for something that resetting the "i" cursor to 0 doesn't?

File: lb/roundrobin.go [hunk 1]
```diff
@@ -111,11 +112,11 @@ func (s *ResultCache) getNextResult(key string) (net.SRV, error) {
 	}
 
 	if len(srvs) <= i {
-		i = 0
+		i = i % len(srvs)
 	}
 
 	srv = srvs[i]
-	s.m[key].i += 1
+	s.m[key].i = i + 1
```
Comment (reterVision) [inline lb/roundrobin.go:119]: The difference is here @benschw , original code set `i` to `s.m[key].i` which leads to a situation that after `i` reaches the largest index, `srv` will only get the first value in the array.

**Comment by benschw** [comment]
10-4, thanks!

**Pull request merged**